### PR TITLE
Fix/select closest available sku from query variant

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

When visiting an URL with variant properties (`propert__{variantName}`) in the query string, we are selecting the first matching SKU instead of the first matching **available** SKU. The first thing the client sees is the unavailable text, so they don't even try to understand that needs to be clicked on the size to see it change.

#### How to test it?

Compare by just opening these pages

- https://kiwi--footnotes.myvtex.com/maci-suede-open-toe-espadrille-275central-maci/p?property__Color=Taupe
-https://footnotes.myvtex.com/maci-suede-open-toe-espadrille-275central-maci/p?property__Color=Taupe

#### Screenshots or example usage:

**After / before**
image

#### Related to / Depends on

https://vtex.slack.com/archives/C9P4RMW6Q/p1595338762403600

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,7 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [0.6.0] - 2019-12-27
 ### Added
-- Docs buidler.
+- Docs builder.
```
Comment (victorhmp) [inline CHANGELOG.md:26]: 🦅 👀

File: react/modules/items.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+export function itemHasVariation(
+  item: Item,
+  variation: { name: string; value: string }
+) {
+  const { name, value } = variation
+
+  return Boolean(
+    item.variations.find(
+      v =>
```
Comment (victorhmp) [inline react/modules/items.ts:15]: I know it's pretty verbose, but it might be better to name this `variation`.

File: react/reducer/index.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { useReducer } from 'react'
-import { path, find, propEq, compose, flip, gt, pathOr } from 'ramda'
+
```
Comment (victorhmp) [inline react/reducer/index.ts:2]: :satisfaction:

File: react/modules/skuQueryString.ts [hunk 4]
```diff
@@ -1,63 +1,47 @@
...
     })
   )
 
-  return selectedItem && selectedItem.itemId
+  const availableItem = itemsWithVariation.find(item =>
+    findAvailableProduct(item)
+  )
+
+  if (availableItem) {
+    return availableItem?.itemId
```
Comment (victorhmp) [inline react/modules/skuQueryString.ts:43]:
```suggestion
    return availableItem.itemId
```

File: react/reducer/index.ts (hunk 3 above)
Comment (kaisermann) [reply react/reducer/index.ts:2]: 👯

**Pull request merged**